commit 4c1540293e0accdf0222f7e822012177d7ca9586 (refs/changes/67/1267/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-01-17T14:37:35-08:00 (2 years, 9 months ago)
    
    joyent/node-triton#167 test suite: use 'nodetritontest' in test key name, `createTestInst` never reports failure

diff --git a/test/integration/cli-keys.test.js b/test/integration/cli-keys.test.js
index 172dbcb..a7482da 100644
--- a/test/integration/cli-keys.test.js
+++ b/test/integration/cli-keys.test.js
@@ -23,7 +23,7 @@ var backoff = require('backoff');
 var KEY_PATH = __dirname + '/data/id_rsa.pub';
 var KEY_SIG  = '66:ca:1c:09:75:99:35:69:be:91:08:25:03:c0:17:c0';
 var KEY_EMAIL = 'test@localhost.local';
-var KEY_NAME = 'testkey';
+var KEY_NAME = 'nodetritontest-key1';
 var MAX_CHECK_KEY_TRIES = 10;
 
 // --- Tests
diff --git a/test/integration/helpers.js b/test/integration/helpers.js
index f40ab56..0e096fc 100644
--- a/test/integration/helpers.js
+++ b/test/integration/helpers.js
@@ -265,13 +265,25 @@ function createClient(cb) {
 function createTestInst(t, name, cb) {
     getTestPkg(t, function (err, pkgId) {
         t.ifErr(err);
+        if (err) {
+            cb(err);
+            return;
+        }
 
         getTestImg(t, function (err2, imgId) {
             t.ifErr(err2);
+            if (err2) {
+                cb(err2);
+                return;
+            }
 
             var cmd = f('instance create -w -n %s %s %s', name, imgId, pkgId);
             triton(cmd, function (err3, stdout) {
                 t.ifErr(err3, 'create test instance');
+                if (err3) {
+                    cb(err3);
+                    return;
+                }
 
                 var match = stdout.match(/Created .+? \((.+)\)/);
                 var inst = match[1];
